client/blocks/app-banner/utils.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Internal dependencies
+ */
+import {
+	get,
+	includes,
+	reduce,
+} from 'lodash';
+
+export const PREFERENCE_NAME = 'appBannerDismissTimes';
+export const EDITOR = 'post-editor';
+export const NOTES = 'notifications';
+export const READER = 'reader';
+export const STATS = 'stats';
+export const ALLOWED_SECTIONS = [ EDITOR, NOTES, READER, STATS ];
+export const ONE_WEEK_IN_MILLISECONDS = 604800000;
+export const ONE_YEAR_IN_MILLISECONDS = 31540000000;
+
+export function getAppBannerData( translate, sectionName ) {
+	switch ( sectionName ) {
+		case EDITOR:
+			return {
+				title: translate( 'Rich mobile publishing.' ),
+				copy: translate( 'A streamlined editor with faster, simpler image uploading? Check and mate.' ),
+			};
+		case NOTES:
+			return {
+				title: translate( 'Watch engagement happening.' ),
+				copy: translate( 'Is your new post a hit? With push notifications, see reactions as they roll in.' ),
+			};
+		case READER:
+			return {
+				title: translate( 'Read online or off.' ),
+				copy: translate( 'Catch up with new posts when you have time, even if you are offline.' ),
+			};
+		case STATS:
+			return {
+				title: translate( 'Stats at your fingertips.' ),
+				copy: translate( "Add real-time stats to your device's notifications or widgets." ),
+			};
+		default:
+			return {
+				title: '',
+				copy: '',
+			};
+	}
+}
+
+export function getCurrentSection( currentSection, isNotesOpen ) {
+	if ( isNotesOpen ) {
+		return NOTES;
+	}
+
+	if ( includes( ALLOWED_SECTIONS, currentSection ) ) {
+		return currentSection;
+	}
+
+	return null;
+}
+
+export function getNewDismissTimes( dismissedSection ) {
+	const currentTime = Date.now();
+	const aWeekFromNow = currentTime + ONE_WEEK_IN_MILLISECONDS;
+	const aYearFromNow = currentTime + ONE_YEAR_IN_MILLISECONDS;
+
+	return reduce( ALLOWED_SECTIONS, ( result, section ) => {
+		result[ section ] = ( section === dismissedSection ) ? aYearFromNow : aWeekFromNow;
```

Comment:
How about adding some randomization to this count? I think it would be more annoying if I were greeted every Monday morning at tea time by such an advert than if it came at some random moment approximately a week after each appearance.

Of course, a week still seems frequent. Would be nice to dismiss for a couple of weeks to a month at a minimum.

```js
// basic Box-Muller uniform -> normal transform
// cite: https://stackoverflow.com/a/36481059/486538
const randomNormal = ( mu, sigma ) => {
	const u = 1 - Math.random(); // Subtraction to flip [0, 1) to (0, 1].
	const v = 1 - Math.random();
	const r = Math.sqrt( -2.0 * Math.log( u ) ) * Math.cos( 2.0 * Math.PI * v );

	return mu + r * sigma;
}

// within a day or two of a week from now
const nagSooner = currentTime + randomNormal( ONE_WEEK_IN_MILLISECONDS, DAY_IN_MILLISECONDS )

// within a month or so of a year from now
const nagLater = currentTime + randomNormal( ONE_YEAR_IN_MILLISECONDS, 30 * DAY_IN_MILLISECONDS )
```